src/components/PopupWithForm.js

```diff
@@ -20,14 +22,48 @@ export default class PopupWithForm extends Popup {
     super.setEventListeners();
     this._form.addEventListener("submit", (event) => {
       event.preventDefault();
-      this._handleFormSubmit(this._getInputValues());
+      this._renderLoading(true); // Show "Saving..." text
+      this._handleFormSubmit(this._getInputValues())
+        .then(() => {
+          this._renderLoading(false); // Reset to original text
```

Comment:
You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`.

No need to repeat it 2 times